File: java/Crud.java
/* Saya Amida Zulfa Laila NIM 2101147 mengerjakan Latihan Praktikum 1
dalam mata kuliah Desain dan Pemrograman Berorientasi Objek untuk keberkahanNya maka
saya tidak melakukan kecurangan seperti yang telah dispesifikasikan.
Aamiin. */
import java.util.ArrayList;
import java.util.Scanner;

/* Deklarasi Class Crud */
public class Crud {
    /* Atribut Private */
    private ArrayList<Mahasiswa> llist; //deklarasi list
    private String nama;
    private int nim;
    private String prodi;
    private String fakultas;
    
    /* Constructor */
    public Crud(){
        llist = new ArrayList<Mahasiswa>();
    }

    /* Method Create atau menambah data ke list */
    public void tambah(Mahasiswa mhs){
        //tambahkan ke list
        llist.add(mhs);
    }

    /* Method hapus atau delete untuk menghapus data pada list */
    public void hapus(int hapus){
        for ( int i = 0; i < llist.size(); i++) {
            if (llist.get(i).getNim()==hapus) {
                llist.remove(i);
            }
        }
    }

    /* Method update untuk mengupdate data dari list */
    public void update(int cari){
        for ( int i = 0; i < llist.size(); i++) {
            if (llist.get(i).getNim()==cari) {
                Scanner sc = new Scanner(System.in);
                System.out.println("--Masukan Nama--");
                nama=sc.next();
                
                System.out.println("--Masukan NIM--");
                nim=sc.nextInt();
                System.out.println("--Masukan Prodi--");
                prodi=sc.next();
                System.out.println("--Masukan Fakultas--");
                fakultas=sc.next();

                llist.get(i).setNama(nama);
                llist.get(i).setNim(nim);
                llist.get(i).setProdi(prodi);
                llist.get(i).setFakultas(fakultas);
                //sc.close();
            }
        }
    }

    /* Method untuk menampilkan seluruh data */
    public void show(){
        System.out.println("============Daftar Mahasiswa============");
        for ( int i = 0; i < llist.size(); i++) {
            System.out.println(Integer.toString(i+1) + ". " + llist.get(i).getNama() + " | " + llist.get(i).getNim()+ " | " + llist.get(i).getProdi()+ " | " + llist.get(i).getFakultas());
        }
    }
}

File: php/Crud.php
<!-- Saya Amida Zulfa Laila NIM 2101147 mengerjakan Latihan Praktikum 1
dalam mata kuliah Desain dan Pemrograman Berorientasi Objek untuk keberkahanNya maka
saya tidak melakukan kecurangan seperti yang telah dispesifikasikan.
Aamiin. -->
<?php
// require file Mahasiswa atau import file Mahasiswa
require ('Mahasiswa.php');
    // class declaration
    class Crud{
        // atribut private 
        private $arrMhs = array(); //deklarasi array
        private $nama='';
        private $nim='';
        private $prodi='';
        private $fakultas='';
        private $foto='';

        // constructor
        public function __construct( $nama = '', $nim ='', $prodi='', $fakultas='', $foto='' ) {
            $this->nama = $nama;
            $this->nim = $nim;
            $this->prodi = $prodi;
            $this->fakultas = $fakultas;
            $this->foto = $foto;
        }

        // public method tambah, untuk create atau menambahkan data pada array
        public function tambah($nama, $nim , $prodi, $fakultas, $foto){
            $mhs = new Mahasiswa($nama , $nim, $prodi, $fakultas, $foto);
            array_push($this->arrMhs, $mhs);
        }

        // public method  untuk menampilkan seluruh data
        public function show()
        {
        $nomor=1;
        echo "<center><table border='1' width='50%' height='50%'>";
        echo "<tr align='center'>
            <th width='10%'> No </th>
            <th width='25%'> Nama </th>
            <th width='25%'> NIM </th>
            <th width='25%'> Program Studi </th>
            <th width='25%'> Fakultas </th>
            <th width='25%'> Foto </th>
            </tr>";
        for($x = 0; $x < sizeof($this->arrMhs); $x++){
            echo "<tr  align='center'>";
            echo '<td>'.$nomor.'</td><td>';
            echo $this->arrMhs[$x]->getNama();
            echo "</td><td>";
            echo $this->arrMhs[$x]->getNim();
            echo "</td><td>";
            echo $this->arrMhs[$x]->getProdi();
            echo "</td><td>";
            echo $this->arrMhs[$x]->getFakultas();
            echo "</td><td>";
            echo "<img src='" . $this->arrMhs[$x]->getFoto(). "'style='width: 100px'>";
            echo "</td>";
            echo "</tr>";
            $nomor++;
        }
            echo "</table> </center>";
        }

        // public method hapus, untuk delete atau menghapus data pada array
        public function hapus($nimHps){
            for ($i=0; $i < sizeof($this->arrMhs); $i++) { 
                if ($this->arrMhs[$i]->getNim()==$nimHps) {
                    array_splice($this->arrMhs,$i,1);
                }
            }
        }

        // public method update, untuk edit atau mengupdate data pada array
        public function update($nimUpdate, $nama, $nim, $prodi, $fakultas, $foto){
            for ($i=0; $i < sizeof($this->arrMhs); $i++) {
                if ($this->arrMhs[$i]->getNim()==$nimUpdate) {
                    $this->arrMhs[$i]->setNama($nama);
                    $this->arrMhs[$i]->setNim($nim);
                    $this->arrMhs[$i]->setProdi($prodi);
                    $this->arrMhs[$i]->setFakultas($fakultas);
                    $this->arrMhs[$i]->setFoto($foto);
                }
            }
        }
    }

?>